### PR TITLE
feature(raw): Improve prisma.raw by re-exporting all of sql-template-tag

### PR DESCRIPTION
**Pull request by zachasme**
When sending raw queries using `prisma.raw` it would be nice to have access to all of `sql-template-tag`.

We had a specific query where we wanted to write the SQL by hand. But the current API is quite limited. Nesting and joining in particular from `sql-template-tag` are very useful when composing complex queries.

This PR simply re-exports the same API as a direct import, adds a case for `prisma.sql` input to `prisma.raw`, and adds a simple test as well.

### CODE REVIEW

**Comment by Sytten** [comment]
Did you manage to get some SQL syntax highlight at the same time? This is something that I would like to see improved.

File: src/packages/client/src/runtime/getPrismaClient.ts [hunk 1]
```diff
@@ -123,6 +123,14 @@ export interface GetPrismaClientOptions {
   internalDatasources: InternalDatasource[]
 }
 
+// Re-export sql-template-tag
+const sqlTemplateTagExport = (
+  ...args: Parameters<typeof sqlTemplateTag.sqltag>
+) => sqlTemplateTag.sqltag(...args)
+sqlTemplateTagExport.empty = sqlTemplateTag.empty
+sqlTemplateTagExport.join = sqlTemplateTag.join
+sqlTemplateTagExport.raw = sqlTemplateTag.raw
```
Comment (Sytten) [inline src/packages/client/src/runtime/getPrismaClient.ts:132]: I would prefer if we could import directly those tags from prisma instead of using the instance to reach them.

File: src/packages/client/src/runtime/getPrismaClient.ts [hunk 2]
```diff
@@ -307,6 +315,12 @@ export function getPrismaClient(config: GetPrismaClientOptions): any {
       }
       return this.disconnectionPromise
     }
+
+    /**
+     * Re-export of 'sql-template-tag'
+     */
+    sql = sqlTemplateTagExport
```
Comment (Sytten) [inline src/packages/client/src/runtime/getPrismaClient.ts:322]: You need to add this in `TSClient.ts` in the folder `generation` so the typing follows.

**Comment by zachasme** [comment]
Thanks for the feedback, I've rebased on master, moved the exports to the module directly (instead of `prisma instance`) and added exports to `TSClient.ts`.

File: src/packages/client/src/runtime/getPrismaClient.ts (hunk 2 above)
Comment (zachasme) [reply src/packages/client/src/runtime/getPrismaClient.ts]: Thanks, I think I've added the correct exports, please let me know. :-)

File: src/packages/client/src/runtime/getPrismaClient.ts [hunk 3]
```diff
@@ -335,9 +336,16 @@ export function getPrismaClient(config: GetPrismaClientOptions): any {
           values: JSON.stringify(queryInstance.values),
           __prismaRawParamaters__: true,
         }
-      } else {
+      } else if ("string" === typeof stringOrTemplateStringsArray) {
         // Called with prisma.raw(string)
         query = stringOrTemplateStringsArray
```
Comment (Sytten) [inline src/packages/client/src/runtime/getPrismaClient.ts:341]: Here I would also add the parameters since someone could write `.raw(string with $1, values)`. This is unrelated to this PR though

File: src/packages/client/src/runtime/getPrismaClient.ts [hunk 4]
```diff
@@ -335,9 +336,16 @@ export function getPrismaClient(config: GetPrismaClientOptions): any {
...
+      } else {
+        // called with prisma.raw(prisma.sql\`\`)
+        query = stringOrTemplateStringsArray[sqlOutput]
+        parameters = {
+          values: JSON.stringify(stringOrTemplateStringsArray.values),
+          __prismaRawParamaters__: true,
```
Comment (Sytten) [inline src/packages/client/src/runtime/getPrismaClient.ts:347]: Typo here

Comment (zachasme) [reply src/packages/client/src/runtime/getPrismaClient.ts:347]: Hm, which part? :)

Comment (Sytten) [reply src/packages/client/src/runtime/getPrismaClient.ts:347]: Paramaters -> Parameters

File: src/packages/client/src/runtime/getPrismaClient.ts (hunk 3 above)
Comment (zachasme) [reply src/packages/client/src/runtime/getPrismaClient.ts:341]: Will do

File: src/packages/client/src/runtime/getPrismaClient.ts (hunk 4 above)
Comment (zachasme) [reply src/packages/client/src/runtime/getPrismaClient.ts:347]: Ah, this is the actual name used in the codebase.

https://github.com/prisma/prisma/blob/788f9e5b1ca024a27d24a4eb8380eb87f7b474b6/src/packages/client/src/runtime/query.ts#L684

You want this handled in another PR or this one?

Comment (Sytten) [reply src/packages/client/src/runtime/getPrismaClient.ts:353]: I believe another PR would be better. Getting this one merged and documented would be a big win, I really need it... 
@timsuchanek @janpio

Comment (janpio) [reply src/packages/client/src/runtime/getPrismaClient.ts:353]: Separate PR sounds better, indeed.

**Comment by zachasme** [comment]
What else can I do to move this along?

**Comment by timsuchanek** [comment]
@zachasme thanks, so far there's nothing needed from your side.
We first need to agree internally, if this is the API, that we want.
As soon as we've done so, we'll let you know here in the PR.

**Comment by timsuchanek** [comment]
Thanks for the PR! I did a few changes about where to import stuff from, it's now building in our CI and will be available on alpha in 10 min.

**Comment by Sytten** [comment]
Wow thanks a lot guys! Good job on landing this
